app/models/subnet.rb

```diff
@@ -185,10 +185,9 @@ def unused_ip(mac = nil, excluded_ips = [])
   end
 
   def known_ips
-    ips = self.interfaces.map(&:ip) + self.hosts.includes(:interfaces).map(&:ip)
-    ips += [self.gateway, self.dns_primary, self.dns_secondary].select(&:present?)
+    ips = [self.gateway, self.dns_primary, self.dns_secondary].select(&:present?)
     self.clear_association_cache
-    ips.compact.uniq
+    ips
   end
```

Comment:
This does not seem to be exercised, mind adding an extra unit test for each subclass?